cli/cmd/sync2.go

```diff
@@ -0,0 +1,148 @@
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create destination plugin client for %s: %w", destinationSpec.Name, err)
+		}
+		if err := destClients[i].Initialize(ctx, destinationSpec); err != nil {
+			return fmt.Errorf("failed to initialize destination plugin client for %s: %w", destinationSpec.Name, err)
+		}
+
+		if err := destClients[i].Migrate(ctx, tables); err != nil {
+			return fmt.Errorf("failed to migrate source %s on destination %s : %w", sourceSpec.Name, destinationSpec.Name, err)
+		}
+	}
+
+	resources := make(chan []byte)
+	g, gctx := errgroup.WithContext(ctx)
+	log.Info().Str("source", sourceSpec.Name).Strs("destinations", sourceSpec.Destinations).Msg("Start fetching resources")
+	fmt.Println("Starting sync for: ", sourceSpec.Name, "->", sourceSpec.Destinations)
+	g.Go(func() error {
+		defer close(resources)
+		if err := sourceClient.Sync2(gctx, sourceSpec, resources); err != nil {
+			if isUnknownConcurrencyFieldError(err) {
+				return fmt.Errorf("unsupported version of source %s@%s. Please update to the latest version from https://cloudquery.io/docs/plugins/sources", sourceSpec.Name, sourceSpec.Version)
+			}
+			return fmt.Errorf("failed to sync source %s: %w", sourceSpec.Name, err)
+		}
+		return nil
+	})
+
+	bar := progressbar.NewOptions(-1,
+		progressbar.OptionSetDescription("Syncing resources..."),
+		progressbar.OptionSetItsString("resources"),
+		progressbar.OptionShowIts(),
+		progressbar.OptionSetElapsedTime(true),
+		progressbar.OptionShowCount(),
+		progressbar.OptionClearOnFinish(),
+	)
+	failedWrites := uint64(0)
+	totalResources := uint64(0)
+	for i, destination := range sourceSpec.Destinations {
+		i := i
+		destination := destination
+		g.Go(func() error {
+			var destFailedWrites uint64
+			var err error
+			if err = destClients[i].Write2(gctx, tables, sourceSpec.Name, syncTime, destSubscriptions[i]); err != nil {
+				return fmt.Errorf("failed to write for %s->%s: %w", sourceSpec.Name, destination, err)
+			}
+			if destClients[i].Close(ctx); err != nil {
+				return fmt.Errorf("failed to close destination client for %s->%s: %w", sourceSpec.Name, destination, err)
+			}
+			failedWrites += destFailedWrites
+			return nil
+		})
+	}
+
+	g.Go(func() error {
+		for resource := range resources {
+			totalResources++
+			_ = bar.Add(1)
+			for i := range destSubscriptions {
+				select {
+				case <-gctx.Done():
+					return gctx.Err()
+				case destSubscriptions[i] <- resource:
+				}
+			}
+		}
+		for i := range destSubscriptions {
+			close(destSubscriptions[i])
+		}
+		return nil
+	})
+
+	if err := g.Wait(); err != nil {
+		_ = bar.Finish()
+		return err
+	}
+
+	_ = bar.Finish()
+
+	metrics, err := sourceClient.GetMetrics(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get metrics for source %s: %w", sourceSpec.Name, err)
+	}
+
+	fmt.Printf("Sync completed successfully. Resources: %d, Errors: %d, Panics: %d\n", metrics.TotalResources(), metrics.TotalErrors(), metrics.TotalPanics())
```

Comment:
Looks like we're missing the time taken now, which I think was quite nice to have.

Reply:
Added